File: android/src/main/kotlin/org/perfect/notifications/perfect_notifications/service/LogService.kt
package org.perfect.notifications.perfect_notifications.service

import android.util.Log

object LogService {
    fun debug(message: String, tag: String = "") {
        Log.d("🤖 Perfect Notifications: $tag : ", "\n$message")
    }

    fun info(message: Any, tag: String = "") {
        Log.i("ℹ️ Perfect Notifications: $tag : ", "\n$message")
    }

    fun success(message: Any, tag: String = "") {
        Log.i("✅ Perfect Notifications: $tag : ", "\n$message")
    }

    fun warning(message: Any, tag: String = "") {
        Log.w("⚠️ Perfect Notifications: $tag : ", "'\n$message")
    }

    fun error(error: Exception, tag: String = "") {
        Log.e("❌ Perfect Notifications: $tag : ", "\nmessage : ${error.message}")
    }
}

File: ios/Classes/src/service/LogService.swift
import Foundation

class LogService {

    static func debug(_ message: String, tag: String = "") {
        NSLog("🤖 Perfect Notifications [\(tag)] : \n\(message)")
    }

    static func info(_ message: Any, tag: String = "") {
        NSLog("ℹ️ Perfect Notifications [\(tag)] : \n\(message)")
    }

    static func success(_ message: Any, tag: String = "") {
        NSLog("✅ Perfect Notifications [\(tag)] : \n\(message)")
    }

    static func warning(_ message: Any, tag: String = "") {
        NSLog("⚠️ Perfect Notifications [\(tag)] : \n\(message)")
    }

    static func error(_ error: String, tag: String = "") {
        NSLog("❌ Perfect Notifications [\(tag)] : \nmessage: \(error)")
    }
}
